packages/api/cms-api/src/auth/auth.resolver.ts

```diff
@@ -18,3 +21,32 @@ export function createAuthAuthedUserResolver<CurrentUser>(CurrentUser: Type<Curr
     }
     return AuthedUserResolver;
 }
+
+export function createAuthJwtResolver<CurrentUser>(CurrentUser: Type<CurrentUser>): Type<unknown> {
+    @Resolver(() => CurrentUser)
+    class JwtAuthResolver {
+        constructor(@Inject(forwardRef(() => AUTH_JWT_CONFIG)) private config: JwtConfig) {}
+
+        @Query(() => CurrentUser)
+        async me(@GetCurrentUser() user: typeof CurrentUser): Promise<typeof CurrentUser> {
+            return user;
+        }
+
+        @Mutation(() => String)
+        async signOut(@Context("req") req: IncomingMessage): Promise<string | null> {
+            let redirectUri = this.config.postLogoutRedirectUri || "/";
+
+            if (req.headers["authorization"] && this.config.endSessionEndpoint) {
+                const url = new URL(this.config.endSessionEndpoint);
+                url.search = new URLSearchParams({
+                    id_token_hint: req.headers["authorization"].substring(7),
+                    post_logout_redirect_uri: redirectUri,
+                }).toString();
+                redirectUri = url.toString();
+            }
+
+            return `/oauth2/sign_out?rd=${encodeURIComponent("/")}`;
```

Comment:
returning a string doesn't make it obvious that this is a url that should be redirected to

Reply:
The resolver has to have the same signature like in the AuthedUser-Resolver to be able to use a generic Logout-Button in the application. The hardcoded /oauth2/sign_out?rd= is not nice though, do you have a better idea?

Reply:
>  The hardcoded /oauth2/sign_out?rd= is not nice though, do you have a better idea?

You could add it to the options of the factory and provide a default value. Is `?rd=` standard? Otherwise we could also add a `buildRedirectUrl` function to the factory's options

Reply:
`?rd` is not standard. Indeed I could add it to the options, but how do I transport the information that the redirectUri has to be encoded? I can't find a clean solution, especially because this is something which should be handled by the proxy: https://github.com/oauth2-proxy/oauth2-proxy/issues/1656

Reply:
See https://github.com/vivid-planet/comet/pull/920/commits/ee3107546e4c4501ca58a6f01e2a73c9ac6e7531